reference/src/client/include/sa_ta_types.h

```diff
@@ -189,38 +189,41 @@ typedef struct {
 
 typedef struct {
     uint8_t iv[GCM_IV_LENGTH];
-    size_t iv_length;
+    uint64_t iv_length;
     uint8_t tag[MAX_GCM_TAG_LENGTH];
-    uint8_t tag_length;
+    uint64_t tag_length;
 } sa_unwrap_parameters_aes_gcm_s;
 
 typedef struct {
     uint8_t counter[CHACHA20_COUNTER_LENGTH];
-    size_t counter_length;
+    uint64_t counter_length;
     uint8_t nonce[CHACHA20_NONCE_LENGTH];
-    size_t nonce_length;
+    uint64_t nonce_length;
 } sa_unwrap_parameters_chacha20_s;
 
 typedef struct {
     uint8_t nonce[CHACHA20_NONCE_LENGTH];
-    size_t nonce_length;
+    uint64_t nonce_length;
     uint8_t tag[CHACHA20_TAG_LENGTH];
-    uint8_t tag_length;
+    uint64_t tag_length;
 } sa_unwrap_parameters_chacha20_poly1305_s;
 
 typedef struct {
-    sa_digest_algorithm digest_algorithm;
-    sa_digest_algorithm mgf1_digest_algorithm;
+    uint32_t digest_algorithm;
```

Comment:
What's the benefit to changing these to 32-bit unsigned integers?

Reply:
Enums are typically represented as uint32_t. This is just explicitly declaring it.